packages/common/src/api/tan-query/useAudioBalance.ts

```diff
@@ -0,0 +1,176 @@
+  includeStaked,
+  chain
+}: UseWalletAudioBalanceParams) => [
+  QUERY_KEYS.audioBalance,
+  chain,
+  address,
+  includeStaked
+]
+
+const fetchWalletAudioBalance = async (
+  { sdk, env }: { sdk: AudiusSdk; env: Env },
+  { address, includeStaked, chain }: UseWalletAudioBalanceParams
+) => {
+  if (chain === Chain.Eth) {
+    const checksumWallet = getAddress(address)
+    const balance = await sdk.services.audiusTokenClient.balanceOf({
+      account: checksumWallet
+    })
+    if (!includeStaked) {
+      return AUDIO(balance).value
+    }
+    const delegatedBalance =
+      await sdk.services.delegateManagerClient.getTotalDelegatorStake({
+        delegatorAddress: checksumWallet
+      })
+    const stakedBalance = await sdk.services.stakingClient.totalStakedFor({
+      account: checksumWallet
+    })
+
+    return AUDIO(balance + delegatedBalance + stakedBalance).value
+  } else {
+    const connection = sdk.services.solanaClient.connection
+    const associatedTokenAccountAddress = getAssociatedTokenAddressSync(
+      new PublicKey(env.WAUDIO_MINT_ADDRESS),
+      new PublicKey(address)
+    )
+    const associatedTokenAccount = await getAccount(
+      connection,
+      associatedTokenAccountAddress
+    )
+    return AUDIO(wAUDIO(associatedTokenAccount.amount)).value
+  }
+}
+
+/**
+ * Query function for getting the AUDIO balance of an Ethereum or Solana wallet.
+ */
+export const useWalletAudioBalance = (
+  { address, includeStaked, chain }: UseWalletAudioBalanceParams,
+  options?: QueryOptions
+) => {
+  const { audiusSdk, env } = useAudiusQueryContext()
+
+  return useQuery({
+    queryKey: getWalletAudioBalanceQueryKey({ address, includeStaked, chain }),
+    queryFn: async () => {
+      const sdk = await audiusSdk()
+      return await fetchWalletAudioBalance(
+        { sdk, env },
+        { address, includeStaked, chain }
+      )
+    },
+    ...options
+  })
+}
+
+type UseAudioBalancesParams = {
+  wallets: Array<{ address: string; chain: Chain }>
+  includeStaked?: boolean
+}
+
+/**
+ * Query function for getting the AUDIO balance of several Ethereum or Solana wallets.
+ */
+export const useWalletAudioBalances = (
+  params: UseAudioBalancesParams,
+  options?: QueryOptions<AudioWei>
+) => {
+  const { audiusSdk, env } = useAudiusQueryContext()
+  return useQueries({
+    queries: params.wallets.map(({ address, chain }) => ({
+      queryKey: getWalletAudioBalanceQueryKey({
+        address,
+        chain,
+        includeStaked: true
+      }),
+      queryFn: async () => {
+        const sdk = await audiusSdk()
+        return await fetchWalletAudioBalance(
+          { sdk, env },
+          {
+            address,
+            chain,
+            includeStaked: true
+          }
+        )
+      },
+      ...options
+    }))
+  })
+}
+
+/**
+ * Hook for getting the AUDIO balance of the current user, including connected wallets.
+ *
+ * NOTE: Does not stay in sync with the store. Won't reflect optimism.
```

Comment:
We have some functions (eg tipping, etc) in the app that call a special action to modify balances optimistically that wouldn't be represented by consumers of this hook. For now, nothing consumes this hook (I undid those changes) but left this in as a reference for when we get there.

Reply:
We have to use it everywhere instead of the action. In things like "sendTip", if we want the optimism, we'd have to `setQueryData` on its queryKey or we can just trigger a refetch once the tip is sent. Would be best done in the mutation.

There's a whole web of interdependent things, so probably have to unravel the spiderweb a bit at a time and sync things. I just felt it was out of scope for now.